relay-config/src/config.rs

```diff
@@ -1156,16 +1159,25 @@ impl Config {
         }
 
         match self.values.http.auth_interval {
-            None | Some(0) => None,
-            Some(secs) => Some(Duration::from_secs(secs)),
+            None => None,
+            Some(i) if i <= 0.0 => None,
+            Some(secs) => Some(Duration::from_secs_f64(secs)),
         }
     }
 
-    /// The maximum amount of time that a Relay is allowed to take to re-authenticate with
-    /// the upstream after which it is declared as un-authenticated (if it is not able to
-    /// authenticate).
-    pub fn http_auth_grace_period(&self) -> Duration {
-        Duration::from_secs(self.values.http.auth_grace_period)
+    /// The maximum time of experiencing uninterrupted network failures until Relay considers that
+    /// it has encountered a network outage.
+    pub fn network_outage_grace_period(&self) -> Duration {
+        if self.values.http.network_outage_grace_period <= 0.0 {
+            log::warn!(
```

Comment:
We should not warn here, as this would warn every time we access the value, which would create a lot of very spammy warning messages. Either we warn only once, or even better (see my above comment) we only allow `u64`.

Reply:
Yes I thought about this as well, you will get spammed with warnings but you can easily remedy that by not setting a negative time period. I even thought of panicking if the user miss configures the period so badly.

I'm not sure how to catch this at deserialisation time.